tracing-attributes/src/lib.rs

```diff
@@ -552,8 +553,14 @@ fn gen_block(
         )
     } else {
         quote_spanned!(block.span()=>
-            let __tracing_attr_span = #span;
-            let __tracing_attr_guard = __tracing_attr_span.enter();
+            let __tracing_attr_span;
+            let __tracing_attr_guard;
+            if tracing::level_enabled!(#level) {
+                __tracing_attr_span = #span;
+                __tracing_attr_guard = __tracing_attr_span.enter();
+            }
```

Comment:
It might be nice to have a comment here explaining why we do this? I think to a naive reader who doesn't understand some of the background, this looks like code that could be simplified, so it might be nice to document _why_ it isn't...